### PR TITLE
icon-calendar in public

### PR DESCRIPTION
**Pull request by maho**
Icon-calendar of datefield (and monthfield) should be IMHO by in public/ , not in example only.

### CODE REVIEW

File: library/pyjamas/ui/Calendar.py [hunk 1]
```diff
@@ -394,7 +396,9 @@ def drawNextYear(self):
 
 class DateField(Composite):
 
-    icon_img = "icon_calendar.gif"
+    img_base = pygwt.getImageBaseURL(False)
```
Comment (anthonyrisinger) [inline library/pyjamas/ui/Calendar.py]: i would actually prefer to have all images store themselves in `images`. let this be `True` (ie. just call the method, don't add `True` ;-) and move the `icon_calendar.gif` to `public/images`

File: library/pyjamas/ui/Calendar.py [hunk 2]
```diff
@@ -394,7 +396,9 @@ def drawNextYear(self):
...
+    icon_img = img_base + "/icon_calendar.gif"
```
Comment (anthonyrisinger) [inline library/pyjamas/ui/Calendar.py]: remove the `/` as `pygwt.getImageBaseURL()` will always have a trailing slash (`/` also breaks relative urls)

**Comment by anthonyrisinger** [comment]
this LGTM, just make those couple changes and we'll merge.  btw, the `pygwt` module uses `location.href` to compute the baseurl, and that is indirectly configurable via `<meta name="pygwt:module" content="/path/to/my/module/=MyModule">` see:

https://github.com/pyjs/pyjs/blob/master/pyjs/src/pyjs/builtin/public/bootstrap.js#L95
https://github.com/pyjs/pyjs/blob/master/library/pygwt.browser.py#L17

File: library/pyjamas/ui/Calendar.py (hunk 2 above)
Comment (maho) [inline library/pyjamas/ui/Calendar.py]: W dniu 26.07.2012 08:24, C Anthony Risinger pisze:

> > @@ -394,7 +396,9 @@ def drawNextYear(self):
> > 
> >   class DateField(Composite):
> > -    icon_img = "icon_calendar.gif"
> > -    img_base = pygwt.getImageBaseURL(False)
> > -    icon_img = img_base + "/icon_calendar.gif"
> >   remove the `/` as `pygwt.getImageBaseURL()` will always have a trailing slash (`/` also breaks relative urls)

Ok. I will fix it and then what ... how can I "fix" this pull request 
after I fix it in my clone repo?

## 

pozdrawiam

Łukasz Mach - lukasz.mach@pagema.net

File: library/pyjamas/ui/Calendar.py (hunk 1 above)
Comment (maho) [inline library/pyjamas/ui/Calendar.py]: W dniu 26.07.2012 08:23, C Anthony Risinger pisze:

> > @@ -394,7 +396,9 @@ def drawNextYear(self):
> > 
> >   class DateField(Composite):
> > -    icon_img = "icon_calendar.gif"
> > -    img_base = pygwt.getImageBaseURL(False)
> >   i would actually prefer to have all images store themselves in `images`. let this be `True` (ie. just call the method, don't add `True` ;-) and move the `icon_calendar.gif` to `public/images`

Sounds reasonable.

## 

pozdrawiam

Łukasz Mach - lukasz.mach@pagema.net

**Comment by anthonyrisinger** [comment]
the pull request is tied to the branch in **your** repo. so, to update the pull request, all you have to do is add some commits to your `master` branch -- just use a different/dedicated branch next time to make things easier for yourself :-)

or you can `git reset --hard origin/master`, make a _single_ commit with the changes included, and force push to Github (`git push origin +master:master`)

**Comment by maho** [comment]
W dniu 26.07.2012 09:04, C Anthony Risinger pisze:

> the pull request is tied to the branch in **your** repo. so, to update the pull request, all you have to do is add some commits to your `master` branch -- just use a different/dedicated branch next time to make things easier for yourself :-)
> 
> or you can `git reset --hard origin/master`, make a _single_ commit with the changes included, and force push to Github (`git push origin +master:master`)

I should learn git first, mercurial next, not in reverse order.

I completely don't understand git philosophy :)

## 

pozdrawiam

Łukasz Mach - lukasz.mach@pagema.net

**Comment by anthonyrisinger** [comment]
heh yeah :-) it can seem a little much, but once you realize how massively simple the **very bottom** is ... the terminology makes more sense.

i cannot recommend more checking out http://progit.org and reading the first three chapters at the very least (not long at all) ... it was the "ah-ha!" moment for me a couple years back. the author does a great job breaking it down (and is a Github founder IIRC).

File: library/pyjamas/ui/Calendar.py (hunk 1 above)
Comment (maho) [inline library/pyjamas/ui/Calendar.py]: I need to add True, because False is default.

**Comment by maho** [comment]
Okay, I modified repo according to your notes

**Comment by anthonyrisinger** [comment]
@mahomahomaho, perfect -- thanks!

**Pull request merged**